src/components/Table/TableFormatters/DetailFormatter.vue

```diff
@@ -172,14 +176,19 @@ export default {
       if (this.formatterArgs.beforeClick) {
         this.formatterArgs.beforeClick(this.callbackArgs)
       }
+
       if (this.formatterArgs.onClick) {
-        this.formatterArgs.onClick(this.callbackArgs)
-        return
+        return this.formatterArgs.onClick(this.callbackArgs)
       }
+
       if (this.formatterArgs.drawer) {
-        this.showDrawer()
+        return this.showDrawer()
+      }
+
+      if (this.prevantClick) {
         return
       }
+
       this.goDetail()
     },
     getDetailRoute() {
```

Comment:
I'd need more information about the code snippet you've provided to understand and discuss its differences.